package/greposcope

```diff
@@ -0,0 +1,88 @@
+#!/bin/bash
+
+# SPDX-License-Identifier: GPL-2.0
+
+# Dependencies:
+# rebuilderd-tools
+# parallel
+
+progname="greposcope"
+tmp_dir="${TMPDIR:-/tmp}/${progname}-${UID}"
+rebuilderd_host="https://reproducible.archlinux.org"
+
+help() {
+	cat <<EOF
+Usage: ${progname} [OPTIONS] pattern
+
+Download and search for 'pattern' in diffoscope outputs of every unreproducible packages reported at ${rebuilderd_host}.
+This is useful to identify packages that are unreproducible because of a specific issue.
+
+OPTIONS
+    -h, --help            Show this message
+    -i, --ignore-case     Ignore case distinctions in patterns and input data
+
+Examples:
+    $ ${progname} "gzip compressed data"
+    $ ${progname} -i zipinfo
+EOF
+}
+
+while ((${#})); do
+	case "${1}" in
+		-h|--help)
+			help
+			exit 0
+		;;
+		-i|--ignore-case)
+			extra_grep_opt="--ignore-case"
+		;;
+		--)
+			shift
+			break
+		;;
+		-*)
+			echo -e >&2 "Invalid argument -- '${1}'\nTry '${progname} --help' for more information"
+			exit 1
+		;;
+		*)
+			pattern="${1}"
+		;;
+	esac
+	shift
+done
+
+if [ -z "${pattern}" ]; then
+    echo -e >&2 "No pattern provided\nTry '${progname} --help' for more information"
+    exit 1
+fi
+
+mkdir -p "${tmp_dir}"
+
+echo -e "==> Fetching the list of unreproducible packages...\n"
+mapfile -t pkg_list < <(rebuildctl -H "${rebuilderd_host}" pkgs ls --status BAD | awk '{print $3}')
+
+if [ ${#pkg_list[@]} -eq 0 ]; then
+	echo "==> No unreproducible packages found"
+	exit 0
+fi
+
+updated_pkg_list=()
+for pkg in "${pkg_list[@]}"; do
+    if [ ! -f "${tmp_dir}/${pkg}.diffoscope" ]; then
+        updated_pkg_list+=("${pkg}")
+    fi
+done
+
+pkg_list=("${updated_pkg_list[@]}")
+
+if [ ${#pkg_list[@]} -eq 0 ]; then
```

Comment:
```suggestion
if ! (({#pkg_list[@]})); then
```

I like this form a bit better. But more a suggestion then something you need to change.

Reply:
Fine by me, applied :) https://github.com/archlinux/contrib/pull/89/commits/f869a9ffc4dce1554b0ec0c2025c0a946586081a